pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,373 @@
+                                      containerInfo.ContainerID,
+                                      containerInfo.ContainerNetNS,
+                                      netinfo.InterfaceName,
+                                      containerInfo.Mtu,
+                                      podSriovVFDeviceID[0],
+                                      result,
+                                      ); err != nil {
+                 klog.ErrorS(err, "Failed to configure interfaces for container ", containerInfo.ContainerID, err)
+                 return err
+        }
+	return nil
+}
+
+func (pc *PodController) configureVethPairAsSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, containerInfo *cnipodcache.CNIConfigInfo, result *current.Result) error {
+     return nil
+}
+
+func (pc *PodController) configureSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo, cniconfig []byte) error {
+	var err error
+        var ipamResult *current.Result
+        var cmdArgs *invoke.Args
+
+        for _, containerInfo := range podCNIInfo {
+               cmdArgs = &invoke.Args{Command: string("ADD"), ContainerID: containerInfo.ContainerID,
+                       NetNS: containerInfo.ContainerNetNS, IfName: containerInfo.ContainerIfDev,
+                       Path: cniPath}
+
+               ipamResult, err = ipam.GetIPAMSubnetAddress(cniconfig, cmdArgs)
+               if err != nil {
+                      return errors.New("Secondary network global IPAM failed")
+               }
+
+               result := &current.Result{CNIVersion: containerInfo.CNIVersion}
+               result.IPs = ipamResult.IPs
+               result.Routes = ipamResult.Routes
+	       // Set result.Interface to container interface.
+               for _, ip := range result.IPs {
+                      ip.Interface = current.Int(1)
+               }
+               // If secondary interface name not provided in pod annotation, assign default name.
+               if len(netinfo.InterfaceName) == 0 {
+                    netinfo.InterfaceName = netDev
+               }
+
+               if netinfo.InterfaceType == sriov {
+		    pc.configureSriovAsSecondaryInterface(pod, netinfo, containerInfo, result)
+               } else {
+                    pc.configureVethPairAsSecondaryInterface(pod, netinfo, containerInfo, result)
+               }
+        }
+	return nil
+}
+
+func (pc *PodController) configureSecondaryNetwork(pod *corev1.Pod, networklist []*SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo) error {
+
+        for _, netinfo := range networklist {
+               klog.InfoS("Secondary Network Information:", netinfo)
+               if len(netinfo.NetworkName) > 0 {
+                        netDefCRD, err := pc.netAttachDefClient.NetworkAttachmentDefinitions(pod.Namespace).Get(context.TODO(), netinfo.NetworkName, metav1.GetOptions{})
+                        if err != nil {
```

Comment:
Also add a comment here saying: if NetworkAttachment is not found, we will retry later?